### PR TITLE
Document FMT_STRING requires >C++14

### PR DESCRIPTION
**Pull request by yeswalrus**
Per the title & conversation in https://github.com/fmtlib/fmt/issues/2039, updates the documentation.

### CODE REVIEW

File: doc/api.rst [hunk 1]
```diff
@@ -157,9 +157,12 @@ string checks, wide string, output iterator and user-defined type support.
...
 
-Compile-time checks are enabled when using ``FMT_STRING``. They support built-in
-and string types as well as user-defined types with ``constexpr`` ``parse``
-functions in their ``formatter`` specializations.
+Compile-time checks are supported in C++14 and up. They are enabled when using
+``FMT_STRING`` and support built-in and string types as well as user-defined 
+types with ``constexpr`` ``parse`` functions in their ``formatter`` specializations.
+
+``FMT_STRING`` may function in C++11 environments, but is not supported and can
+produce illegible error messages. It is also known to fail in GCC when passing zero arguments.
```
Comment (vitaut) [inline doc/api.rst:165]: This is no longer the case, please remove.

**Review by vitaut — CHANGES_REQUESTED** [review]
Thanks for the PR

File: doc/api.rst (hunk 1 above)
Comment (yeswalrus) [reply doc/api.rst:165]: https://github.com/fmtlib/fmt/pull/2042 only fixed this for FMT_COMPILE, not FMT_STRING in old gcc versions?

**Comment by vitaut** [comment]
`FMT_STRING` does work with C++11 where it is a noop: https://godbolt.org/z/rfsqzc. If you get an error please open an issue with a repro.

**Pull request closed without merging**